### ISSUE TITLE
setStyleClasses

### ISSUE THREAD

**Issue by robipresotto** [mapbox/mapbox-gl-native#1058]
How is it possible to reset the applied style classes? is it necessary to set the function to nil?
or maybe something like that is missing in function:

mbglMap->removeClass([appliedClass cStringUsingEncoding:[NSString defaultCStringEncoding]]);

the problem is when you set the style classes to night and after change to contrast the system sometimes inverts the styles.

**Comment by jfirebaugh** [comment]
> How is it possible to reset the applied style classes?

Call `setStyleClasses` with an empty array.

> the problem is when you set the style classes to night and after change to contrast the system sometimes inverts the styles.

I don't understand what you're saying here. If you think there's a bug, please open a separate issue with steps to replicate. Thanks!

**Comment by friedbunny** [comment]
@robipresotto Have a look at 4ddae7d1b75cc7d79b08d38a960559a05757d3dc, which does this for the demo app.

**Comment by robipresotto** [comment]
now works better, some days ago the function was jumping the classes many times

**Comment by robipresotto** [comment]
this function does not work properly anymore - last week it is was working perfectly and this week the system chokes and changes only after zooming on the map.
